File: FinanceApp.Shared/Dto/ForecastItem.cs
﻿namespace FinanceApp.Shared.Dto
{
    public class ForecastItem
    {        
        public DateTime DateReference { get; set;}
        public double NominalLiquidValue { get; set;}
        public double RealLiquidValue { get; set;}   
        public double NominalNotLiquidValue { get; set; }
        public double RealNotLiquidValue { get; set; }
        public double NominalCumulatedAmount { get; set;}
        public double RealCumulatedAmount { get; set;}
        public double NominalValue => NominalLiquidValue + NominalNotLiquidValue;
        public double RealValue => RealLiquidValue + RealNotLiquidValue;
    }
}

File: FlutterApp/finance_app/lib/models/forecast/forecast_item.dart
class ForecastItem {
  DateTime dateReference;
  double nominalLiquidValue;
  double nominalNotLiquidValue;
  double nominalValue;
  double realLiquidValue;
  double nominalCumulatedAmount;
  double realCumulatedAmount;

  ForecastItem(
      {required this.dateReference,
      required this.nominalCumulatedAmount,
      required this.nominalValue,
      required this.nominalNotLiquidValue,
      required this.nominalLiquidValue,
      required this.realCumulatedAmount,
      required this.realLiquidValue});

  ForecastItem.fromJson(Map<String, dynamic> json)
      : dateReference = DateTime.parse(json['dateReference']),
        nominalCumulatedAmount = json['nominalCumulatedAmount'].toDouble(),
        nominalLiquidValue = json['nominalLiquidValue'].toDouble(),
        realLiquidValue = json['realLiquidValue'].toDouble(),
        realCumulatedAmount = json['realCumulatedAmount'].toDouble(),
        nominalNotLiquidValue = json['nominalNotLiquidValue'].toDouble(),
        nominalValue = json['nominalValue'].toDouble();
}
